source/Helpers/POETradeAPI/Exchange/ExchangePrefetch.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Sidekick.Helpers.POETradeAPI.Exchange
+{
+    class ExchangePrefetch
+    {
+        public static async Task<ExchangePrefetchResponse> MakeRequest(HttpClient http, Item item, int minimumStock)
+        {
+            var itemName = await TranslateItemName(item);
+            Logger.Log($"ExchangePrefetch: {item.Name}, api name: {itemName}");
+            if (itemName == null)
+            {
+                Logger.Log("ExchangePrefetch: not available for this item");
+                return null;
+            }
+
+            var options = new Dictionary<string, object>
+            {
+                { "status", new Dictionary<string, string > { { "option", "online" } } },
+                { "have", new List<string> { "chaos" } },
+                { "want",  new List<string> { itemName } },
+                { "minimum", minimumStock }
+            };
+            var body = new Dictionary<string, object> { { "exchange", options } };
+            var bodySerialized = new StringContent(JsonConvert.SerializeObject(body), Encoding.UTF8, "application/json");
+
+            // TODO: league is currently hardcoded in URL
+            var response = await http.PostAsync("https://www.pathofexile.com/api/trade/exchange/Metamorph", bodySerialized);
```

Comment:
This should use the selected league, all this information is present inside the static TradeClient class.